Remove use of six.binary_type; use bytes instead

jdufresne opened the pull request.

jdufresne committed:
Remove use of six.binary_type

Remove type checks for cases that don't apply to Python 3.

For remaining uses, use bytes instead

tk0miya commented:
Thank you always :-)

tk0miya reviewed (approved):
Note: I'll merge this after all tests passed.

tk0miya commented:
oh, sorry. I'd failed to merge accidentally.

tk0miya commented on sphinx/cmd/build.py:
```diff
@@ -268,7 +268,7 @@ def build_main(argv=sys.argv[1:]):  # type: ignore
         except ValueError:
             parser.error(__('-D option argument must be in the form name=value'))
-        if likely_encoding and isinstance(val, binary_type):
+        if likely_encoding and isinstance(val, bytes):
```
Does argparse return bytes object in py3? I'll take a look later.

tk0miya commented on sphinx/cmd/quickstart.py:
```diff
@@ -646,7 +646,7 @@ def main(argv=sys.argv[1:]):
     # decode values in d if value is a Python string literal
     for key, value in d.items():
-        if isinstance(value, binary_type):
+        if isinstance(value, bytes):
```
This is also. Does `input()` return bytes object?

tk0miya commented on sphinx/config.py:
```diff
@@ -484,7 +484,7 @@ def check_unicode(config):
+        if isinstance(value, bytes) and nonascii_re.search(value):
             logger.warning(__('the config value %r is set to a string with non-ASCII '
                               'characters; this can lead to Unicode errors occurring. '
                               'Please use Unicode strings, e.g. %r.'), name, u'Content')
```
I think this warning is not meaningful for py3.
This will be warned if users writes binary-string in conf.py like `b'Sphinx'`

tk0miya replied:
Ah, sorry. This function was already deprecated. So no more change is needed.

tk0miya commented on sphinx/ext/apidoc.py:
```diff
@@ -444,13 +442,13 @@ def main(argv=sys.argv[1:]):
+        if isinstance(args.version, bytes):
             d['version'] = d['version'].decode('utf-8')
-        if isinstance(args.release, binary_type):
+        if isinstance(args.release, bytes):
```
Same as quickstart.

tk0miya commented on sphinx/ext/doctest.py:
```diff
@@ -330,7 +330,7 @@ def _warn_out(self, text):
         else:
             logger.info(text, nonl=True)
-        if isinstance(text, binary_type):
+        if isinstance(text, bytes):
```
I don't know that doctest gives bytes object or not. But I suppose this is not needed. binary object is mainly used on IO boundary.

tk0miya commented:
In this morning, I don't have time for review anymore. I'll take a look this again later.
Thank you for your work!

jdufresne commented:
Thanks for the review @tk0miya. I agree with your analysis, those identified type checks are no longer necessary under Python 3. I've rebased and resolved the merge conflicts. I've removed those type checks, which resulted in deprecating `check_unicode()`. All additional feedback is welcome. Thanks!

jdufresne commented:
Rebased to resolve merge conflicts.

tk0miya commented on sphinx/ext/doctest.py:
```diff
@@ -330,9 +330,7 @@ def _warn_out(self, text):
-        if isinstance(text, binary_type):
-            text = force_decode(text, None)
-        self.outfile.write(text)
+            self.outfile.write(text)
```
This line is wrong indented.

tk0miya commented on sphinx/util/__init__.py:
```diff
@@ -458,7 +458,7 @@ def parselinenos(spec, total):
 def force_decode(string, encoding):
```
I think this function will become meaningless. Let's deprecate this in another PR :-)

tk0miya reviewed (changes requested):
LGTM with nits

jdufresne commented:
Good catch. I fixed the indentation.

tk0miya reviewed (approved):
LGTM!

jdufresne commented:
Rebased to resolve merge conflicts.

tk0miya commented:
Merged. Thank you for rebasing :-)
